### ISSUE TITLE
Add warning to narrative docs about optional extras not functioning on python3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8483]
|[<img alt="derwolfe's avatar" src="https://avatars.githubusercontent.com/u/641357?s=50" width="50" height="50">](https://github.com/derwolfe)| @derwolfe reported|
|-|-|
|Trac ID|trac#8483|
|Type|enhancement|
|Created|2016-06-17 14:40:45Z|
|Branch|https://github.com/twisted/twisted/tree/8483-derwolfe-add-extras-warning|

Add a warning to the narrative documentation to warn users that the optional extras don't currently function on python3. Once python3 supports extras again, this warning should be removed.

This came up for me when I was working on a python2/3 application. I setup my application to depend on `twisted[tls]`; `pyOpenSSL` was installed for the python2 env, but not the python3 env. This failed with an import error for `OpenSSL`.

<details><summary>Searchable metadata</summary>

```
trac-id__8483 8483
type__enhancement enhancement
reporter__derwolfe derwolfe
priority__normal normal
milestone__None None
branch__8483_derwolfe_add_extras_warning 8483-derwolfe-add-extras-warning
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1466174445320459 1466174445320459
changetime__1466267475144882 1466267475144882
version__None None
owner__derwolfe derwolfe

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="derwolfe's avatar" src="https://avatars.githubusercontent.com/u/641357?s=50" width="50" height="50">](https://github.com/derwolfe)|@derwolfe removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="derwolfe's avatar" src="https://avatars.githubusercontent.com/u/641357?s=50" width="50" height="50">](https://github.com/derwolfe)<a name="note_2"></a>|@derwolfe commented|
|-|-|

The github PR is located here: https://github.com/twisted/twisted/pull/197

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @derwolfe|
|-|-|

Looks good. Thanks!

Waiting for build results and will merge.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset 30f6966329c857c3631c60aeb420d84d7828e01e
```
#!CommitTicketReference repository="" revision="30f6966329c857c3631c60aeb420d84d7828e01e"
Merge 8483-derwolfe-add-extras-warning: Add warning to docs about optional extras not functioning on py3.

Author: derwolfe
Reviewer: adiroiban
Fixes: #8483
```
